src/libraries/Microsoft.PowerFx.Core/Syntax/UserDefinitions.cs

```diff
@@ -87,6 +87,40 @@ private bool ProcessUserDefinitions(out UserDefinitionResult userDefinitionResul
             return true;
         }
 
+        /// <summary>
+        /// Process user script and returns user defined functions and named formulas.
+        /// </summary>
+        /// <param name="script">User script containing UDFs and/or named formulas.</param>
+        /// <param name="parseCulture">CultureInfo to parse the script.</param>
+        /// <returns>Tuple.</returns>
+        /// <exception cref="InvalidOperationException">Throw if the user script contains errors.</exception>
+        public static (IEnumerable<UserDefinedFunction>, IEnumerable<NamedFormula>) Process(string script, CultureInfo parseCulture)
```

Comment:
Not really sure why we'd need this API wrapping ProcessUserDefinition? Is the error throwing better than returning a UserDefinitionResult and letting the caller handle that it might be erroneous?

Reply:
This is not only wrapping but also unifying a block of code that would be called from multiple places.
The `throw` is following the original UDF implementation. We can discuss and change it.